packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -1651,7 +1651,9 @@ function prepareFreshStack(root: FiberRoot, lanes: Lanes): Fiber {
   }
 
   if (workInProgress !== null) {
-    let interruptedWork = workInProgress.return;
+    let interruptedWork = workInProgressIsSuspended
```

Comment:
I'm not 100% sure this flag means what I think it means.

Reply:
It does, maybe add a one-line comment explaining why this works. I also like your idea to always yield right after the begin phase though it would require more refactoring.